packages/orm/src/decorators/foreignKey.ts

```diff
@@ -0,0 +1,16 @@
+import { AnyModel, ModelDescriptorsRegistry } from "../Model";
+import { fk } from "..";
+
+// The signature should be extended to accept opts
+export function ForeignKey(toModelName: string, relatedName?: string | undefined) {
+	// the internal implementation could be extracted
+	return function <Target extends AnyModel>(target: Target, propertyName: string): void {
```

Comment:
As the comment says, the internal implementation can be extracted and reused between the decorators.